pkg/deploy/registry/deploylog/rest.go

```
@@ -0,0 +1,259 @@
+func (g *podGetter) Get(ctx kapi.Context, name string) (runtime.Object, error) {
+	namespace, ok := kapi.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.NewBadRequest("namespace parameter required.")
+	}
+	return g.podsNamespacer.Pods(namespace).Get(name)
+}
+
+// REST is an implementation of RESTStorage for the api server.
+type REST struct {
+	Deploy         *Client
+	PodGetter      pod.ResourceGetter
+	ConnectionInfo kclient.ConnectionInfoGetter
+	Timeout        time.Duration
+}
+
+// NewREST creates a new REST for DeploymentLogs. It uses three clients: one for configs,
+// one for deployments (replication controllers) and one for pods to get the necessary
+// attributes to assemble the URL to which the request shall be redirected in order to
+// get the deployment logs.
+func NewREST(
+	dn client.DeploymentConfigsNamespacer,
+	rn kclient.ReplicationControllersNamespacer,
+	pn kclient.PodsNamespacer,
+	connectionInfo kclient.ConnectionInfoGetter) *REST {
+
+	return &REST{
+		Deploy:         &Client{rn: rn, dn: dn},
+		PodGetter:      &podGetter{pn},
+		ConnectionInfo: connectionInfo,
+		Timeout:        defaultTimeout,
+	}
+}
+
+// NewGetOptions returns a new options object for deployment logs
+func (r *REST) NewGetOptions() (runtime.Object, bool, string) {
+	return &deployapi.DeploymentLogOptions{}, false, ""
+}
+
+// New creates an empty DeploymentLog resource
+func (r *REST) New() runtime.Object {
+	return &deployapi.DeploymentLog{}
+}
+
+// Get returns a streamer resource with the contents of the deployment log
+func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.Object, error) {
+	deployLogOpts, ok := opts.(*deployapi.DeploymentLogOptions)
+	if !ok {
+		return nil, errors.NewBadRequest("did not get an expected options.")
+	}
+
+	// Fetch deploymentConfig
+	config, err := r.Deploy.GetDeploymentConfig(ctx, name)
+	if err != nil {
+		return nil, errors.NewNotFound("deploymentConfig", name)
```

Why are we masking the actual error?